### PR TITLE
Make `users_external` be able to authenticate against Atlasssian Crowd

### PR DESCRIPTION
**Pull request by hexa2k9**
We're running Atlassian Crowd to Authenticate Users for various Applications. This adds Support to ownCloud 8 (running OC 8.0.2 here) for running Authentication against Atlassian Crowd.

### CODE REVIEW

**Comment by karlitschek** [comment]
@icewind1991 what do you think?

File: user_external/tests/crowd.php [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+	 */
+	private $instance;
+
+	private function getConfig() {
+		return include(__DIR__.'/config.php');
+	}
+
+	function skip() {
+		$config=$this->getConfig();
+		$this->skipUnless($config['imap']['run']);
```
Comment (DeepDiver1975) [inline user_external/tests/crowd.php]: imap ???

**Comment by DeepDiver1975** [comment]
The code itself looks good - but because this is a shipped app I'm not that happy to add this to user_external.

We need to add it to the documentation and *most important* this is then covered by commercial support - @karlitschek or do I miss anything?

File: user_external/tests/crowd.php (hunk 1 above)
Comment (hexa2k9) [inline user_external/tests/crowd.php]: Sorry, it's been copy & paste up until this point. That's fixed in e3bd710.

**Comment by hexa2k9** [comment]
I'd be happy to help if there are any questions regarding Crowd and its REST API used here. The Code however is as bulletproof as I could make it in case Crowd does not respond or throws an error or stuff like that and should give as much information as needed to debug problems with the integration.

I could however modify the integration a little to provide more detailed errors in case Crowd is not setup correctly which is expected to be done correctly which should be covered by "'ERROR: Atlassian Crowd returned Status %s with Message "%s"'" currently.

We're running this patch ever since I've created a pull-request for this without any problems.

**Comment by hexa2k9** [comment]
As it seems this is unlikely to be merged, I'll abandon this Pull-Request.

My branch however will be kept alive for upcoming releases if there are changes to be made as long as we keep using ownCloud.

**Pull request closed without merging**